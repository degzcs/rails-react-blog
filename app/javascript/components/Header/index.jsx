import React from 'react'
import { Link } from 'react-router-dom'

const Header = () => {
  return(
    <div className= 'container'>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#mainNavBar" aria-controls="mainNavBar" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="mainNavBar">
          <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
            <li className="nav-item active">
              <Link className="nav-link" to='/' >
                Home
              </Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to='/blog' >
                Blog
              </Link>
            </li>
          </ul>
          <ul class="navbar-nav mr-auto mt-2 mt-lg-0 justify-content-end">
            <li className="nav-item">
              <Link className="nav-link" to='/login' >
                Login
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  )
}

export default Header
