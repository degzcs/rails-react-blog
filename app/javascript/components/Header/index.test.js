import React from "react";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

import Header from "./index.jsx";

test('checks rendering of Header component', () => {

  const component = renderer.create(
    <StaticRouter location="someLocation">
      <Header />
    </StaticRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
