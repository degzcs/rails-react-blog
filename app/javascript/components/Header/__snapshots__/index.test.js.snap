// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of Header component 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-md fixed-top navbar-light bg-light"
  >
    <div
      className=" container collapse navbar-collapse"
      id="mainNavBar"
    >
      <ul
        className="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/blog"
            onClick={[Function]}
          >
            Blog
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/info"
            onClick={[Function]}
          >
            Info
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
